### PR TITLE
[BSP] Add stm32l412-st-nucleo BSP

### PR DESCRIPTION
**Pull request by luhuadong**
## 拉取/合并请求描述：(PR description)

[
针对 ST 官方 STM32L412 Nucleo 开发板增加 stm32l412-st-nucleo BSP，已在 GCC（Ubuntu）和 MDK5（Windows）环境下进行测试。
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/stm32l412-st-nucleo/board/Kconfig [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+config SOC_STM32L412RB
+    bool
+    select SOC_SERIES_STM32L4
+    select RT_USING_COMPONENTS_INIT
+    select RT_USING_USER_MAIN
+    default y
+
+menu "Onboard Peripheral Drivers"
+	config BSP_USING_STLINK_TO_USART
```
Comment (BernardXiong) [inline bsp/stm32/stm32l412-st-nucleo/board/Kconfig:11]: Please use space for indentation.

File: bsp/stm32/stm32l412-st-nucleo/board/linker_scripts/link.lds [hunk 2]
```diff
@@ -0,0 +1,158 @@
+/*
+ * linker script for STM32L4XX with GNU ld
+ */
+
+/* Program Entry, set to mark it as "used" and avoid gc */
+MEMORY
+{
+    ROM  (rx) : ORIGIN = 0x08000000, LENGTH = 256k /* 256KB flash */
+    RAM1 (rw) : ORIGIN = 0x20000000, LENGTH =  48k /* 48K sram */
+    RAM2 (rw) : ORIGIN = 0x10000000, LENGTH =  16k /* 16K sram */
```
Comment (BernardXiong) [inline bsp/stm32/stm32l412-st-nucleo/board/linker_scripts/link.lds:10]: > MCU：STM32L412RBT6P，主频 80MHz，128KB FLASH ，40KB RAM。

Is it correct?

Comment (luhuadong) [reply bsp/stm32/stm32l412-st-nucleo/board/linker_scripts/link.lds]: Sorry, I forgot to modify this

File: bsp/stm32/stm32l412-st-nucleo/board/Kconfig (hunk 1 above)
Comment (luhuadong) [reply bsp/stm32/stm32l412-st-nucleo/board/Kconfig]: I had used space instead of tab for indentation

**Comment by thread-liu** [comment]
感谢提交，请在以下链接文件中添加 BSP CI 。
https://github.com/RT-Thread/rt-thread/blob/master/.travis.yml

**Comment by SummerLife** [comment]
@BernardXiong 该 PR 可以合并了

**Pull request merged**